Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -196,6 +196,12 @@ - (NSOperationQueue*)_doMainWork {
         didWork = FALSE;
 
         for (int i = 0; i < NSOperationQueuePriority_Count; i++) {
+            [priv->suspendedCondition lock];
+            while (priv->isSuspended) {
+                [priv->suspendedCondition wait];
+            }
```

Comment:
A while loop plus a condition variable (which this is) is the best way to hold for the suspended state. Consider that we signal on `suspendedCondition` for both suspend and resume, and we need to only hold operation processing when `isSuspended` is true.

---
In reply to: [69805759](https://github.com/Microsoft/WinObjC/pull/613#discussion_r69805759) [](ancestors = 69805759)